File: reverse_integer.py
integer = -123

def reverse_integer(integer):
  reverse = 0
  int_to_str = list(str(integer))
  num = ''

  if '-' in int_to_str:
    num = num + int_to_str.pop(0)
    int_to_str.reverse()
    reverse = int(num + ''.join(int_to_str))
  else:
    int_to_str.reverse()
    reverse = int(''.join(int_to_str))

  if reverse.bit_length() < 32:
    return reverse
  else:
    return 0


print(reverse_integer(integer))

File: ruby/reverse_integer.rb
num = -123

def reverse(num)
  int_to_str = num.to_s.split('')
  x = ''
  reverse = nil

  if int_to_str.include?('-')
    x << int_to_str.shift(1)[0]

    x << int_to_str.reverse.join
    reverse = x.to_i
  else
    reverse = int_to_str.reverse.join.to_i
  end

  if reverse.bit_length < 32
    reverse
  else
    0
  end
end

p reverse(num)